src/core/dataset_reader.js

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2022 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { parseXFAPath } from "./core_utils.js";
+import { SimpleXMLParser } from "./xml_parser.js";
+
+class DataSetXMLParser extends SimpleXMLParser {
+  constructor(options) {
+    super(options);
+    this.node = null;
+  }
+
+  onEndElement(name) {
+    const node = super.onEndElement(name);
+    if (node && name === "xfa:datasets") {
+      this.node = node;
+
+      // We don't need anything else, so just kill the parser.
+      throw new Error();
```

Comment:
To ensure that we can always get helpful/useful error messages we should always provide *some* kind of message string for all errors, hence please do something like:
```javascript
throw new Error("Aborting DatasetXMLParser.");
```